Foundation/NSDecimal.swift

```diff
@@ -153,3 +494,30 @@ public func NSDecimalPower(_ result: UnsafeMutablePointer<Decimal>, _ number: Un
 public func NSDecimalMultiplyByPowerOf10(_ result: UnsafeMutablePointer<Decimal>, _ number: UnsafePointer<Decimal>, _ power: Int16, _ roundingMode: NSDecimalNumber.RoundingMode) -> NSDecimalNumber.CalculationError { NSUnimplemented() }
 
 public func NSDecimalString(_ dcm: UnsafePointer<Decimal>, _ locale: AnyObject?) -> String { NSUnimplemented() }
+
+// == Internal (Swifty) functions ==
+
+extension Decimal {
+    fileprivate var isCompact: Bool {
+        get {
+            return _isCompact != 0
+        }
+        set {
+            if newValue {
+                _isCompact = 1
+            } else {
+                _isCompact = 0
+            }
+        }
+    }
+    fileprivate mutating func compact() {
+        if isCompact || isNaN || _length == 0 {
+            return
+        }
+        // FIXME implement the multiply/divide
+        // Divide by 10 as much as possible
+        // Put the non-empty remainder in place
+        // Set the new exponent
+        isCompact = true
```

Comment:
Mainly because the NSDecimalCompact call is invoked from the init(Double) constructor, without which you couldn't test this. However that part of it is still a work in progress, so this could be left out of this patch set for now.